### ISSUE TITLE
Reduce dimension when indexing with number

### ISSUE THREAD

**Issue by constantinpape** [constantinpape/z5#103]
Currently, z5py does not reduce the number of dimensions when indexing with a single index:
```
f = z5py.File('some/path')
ds = f['some/internal/path']  # ds.shape = [100, 100, 100]
x = ds[:, 0]
print(x.shape)
```
will yield `(100, 1, 100)` instead of `(100, 100)` as expected.

Note: just squeezing the output of `__setitem__` by default is not a good solution, because
we still want `x.shape = (100, 1, 100)` for `x = ds[:, 0:1]`.

**Comment by clbarnes** [comment]
How does the C++ work here? Does it write a 1D array into the memory space which numpy then understands how to ravel? If so, I think I've fixed it. If not, it would be fine but there might be another copy involved.

**Comment by constantinpape** [comment]
> Does it write a 1D array into a contiguous memory space which numpy then understands how to ravel?

Yes, I think it does, but I will have another look at this tomorrow and let you know.

I will also try to finally fix the tests to make working with PRs easier.

**Comment by clbarnes** [comment]
I've pushed what I think will fix both this and #102; I'll make a PR and we can see how it turns out in the morning!

I took a defensive strategy so there might be a couple more copies than strictly necessary.

**Comment by constantinpape** [comment]
Fixed by #104 thanks to @clbarnes.
